community/kernel/src/main/java/org/neo4j/kernel/impl/storemigration/participant/StoreMigrator.java

```diff
@@ -170,7 +171,7 @@ public void migrate( File storeDir, File migrationDir, MigrationProgressMonitor.
 
         RecordFormats oldFormat = selectForVersion( versionToMigrateFrom );
         RecordFormats newFormat = selectForVersion( versionToMigrateTo );
-        if ( !oldFormat.hasSameCapabilities( newFormat, CapabilityType.FORMAT ) )
+        if ( upgradeFormatType( oldFormat, newFormat ) || isDifferentCapabilities( oldFormat, newFormat ) )
```

Comment:
Is this condition correct? It seems like `isDifferentCapabilities` may override the `false` returned by the `FormatFamily` check.